src/Quartz/Impl/JobType.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+
+using Quartz.Util;
+
+namespace Quartz.Impl;
+
+/// <summary>
+/// Store the Job Type and backing field name for serialization
+/// </summary>
+public class JobType
+{
+    private string storableTypeName = string.Empty;
+    private Lazy<Type> lazyType = new(() => throw new InvalidOperationException("Type not defined"));
+
+    public JobType()
+    {
+    }
+
+    public JobType(string? typeStorageName)
```

Comment:
I think the logical name would be `typeName`, like it's named in `Type.GetType`?